### PR TITLE
Add qwen vl CPU example 

### PR DESCRIPTION
**Pull request by Ariadne330**
## Description

- Add Qwen-VL-Chat examples for CPU with `chat` API.

### CODE REVIEW

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ def run_optimize_model(repo_id,
         model = optimize_model(model)
         tokenizer = LlamaTokenizer.from_pretrained(model_path, trust_remote_code=True)
     else:
-        model = AutoModelForCausalLM.from_pretrained(model_path, torch_dtype='auto', low_cpu_mem_usage=True)
+        model = AutoModelForCausalLM.from_pretrained(model_path, torch_dtype='auto', low_cpu_mem_usage=True, trust_remote_code=True,use_cache=True) #XXX
         model = optimize_model(model)
         tokenizer = AutoTokenizer.from_pretrained(model_path)
     end = time.perf_counter()
```
Comment (Oscilloscope98) [inline python/llm/dev/benchmark/all-in-one/run.py:274]: Let's raise another PR for all-in-one benchmark fixes. Do not make one PR too big/complex

Comment (Ariadne330) [reply python/llm/dev/benchmark/all-in-one/run.py:274]: Have updated , and this PR now is only for adding example for Qwen-VL-Chat.

**Comment by jason-dai** [comment]
Update table here: https://github.com/intel-analytics/BigDL#verified-models

File: python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Qwen-VL-Chat
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/README.md:1]: Maybe we should change the folder name to qwen-vl.

Same other places: Qwen-VL-Chat model -> Qwen-VL model

File: python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+### 1. Install
+We suggest using conda to manage the Python environment. For more information about conda installation, please refer to [here](https://docs.conda.io/en/latest/miniconda.html#).
+
+After installing conda, create a Python environment for BigDL-LLM:
+```bash
+conda create -n llm python=3.9 # recommend to use Python 3.9
+conda activate llm
+
+pip install --pre --upgrade bigdl-llm[all] # install the latest bigdl-llm nightly build with 'all' option
+pip install accelerate tiktoken einops scipy transformers_stream_generator==0.0.4 peft deepspeed torchvision matplotlib
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/README.md:18]: Here the additional packages should be based on https://github.com/QwenLM/Qwen-VL/blob/master/requirements.txt, which are not the same as you list here :)

Also it is recommended to add a line of comment here `# additional package required for Qwen-VL-Chat to conduct generation`

File: python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/chat.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                        help='The huggingface repo id for the Qwen_VL_Chat model to be downloaded'
+                             ', or the path to the huggingface checkpoint folder')
+    parser.add_argument('--n-predict', type=int, default=32, help='Max tokens to predict')
+    
+    current_path = os.path.dirname(os.path.abspath(__file__))
+    args = parser.parse_args()
+    model_path = args.repo_id_or_model_path
+
+    # Load model
+    model = AutoModelForCausalLM.from_pretrained(model_path, device_map="cpu",  trust_remote_code=True).eval()
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/chat.py:38]: `.eval()` is not needed?

File: python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/chat.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    # With only one line to enable BigDL-LLM optimization on model
+    # Note that currently there is no optimization support 'c_fc' and 'out_proj' moudule, 'modules_to_not_convert' should be set
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/chat.py:41]: We could change the comment to 

For successful BigDL-LLM optimization on Qwen-VL-Chat, skip the 'c_fc' and 'out_proj' modules during optimization.

File: python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/chat.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    model = optimize_model(model, low_bit='sym_int4', modules_to_not_convert=['c_fc', 'out_proj'] , optimize_llm = True)
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/chat.py:42]: arg `optimize_llm` can be removed as it is default to be `True`

File: python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/chat.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    
+    # Load tokenizer
+    tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
+
+    # Session ID
+    session_id = 1
+
+    while True:
+      print('-'*20, 'Session %d' % session_id, '-'*20)
+      image_input = input(f' 请输入你想询问的图片: ')
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/chat.py:55]: Maybe we should use English here :), just for consistency

File: python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/README.md (hunk 2 above)
Comment (Oscilloscope98) [reply python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/README.md:1]: Also maybe `SimSun.tff` and `Session_2.png` should not be added here in the qwen-vl folder

File: python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/README.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+ 请输入你想询问的图片:
+ 请输入你想询问的文字信息: 输出击掌的检测框
+---------- Response ----------
+<ref>击掌</ref><box>(538,513),(588,601)</box>
+
+-------------------- Session 3 --------------------
+ 请输入你想询问的图片: exit
+```
+
+[![Demo.gif](https://i.postimg.cc/0N1QshjQ/Demo.gif)](https://postimg.cc/yDnBhQF4)
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/Qwen-VL-Chat/README.md:74]: Please add Demo.gif to llm-assets readthedocs :)

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+We suggest using conda to manage the Python environment. For more information about conda installation, please refer to [here](https://docs.conda.io/en/latest/miniconda.html#).
+
+After installing conda, create a Python environment for BigDL-LLM:
+```bash
+conda create -n llm python=3.9 # recommend to use Python 3.9
+conda activate llm
+
+pip install --pre --upgrade bigdl-llm[all] # install the latest bigdl-llm nightly build with 'all' option
+
+pip install transformers==4.31.0 accelerate tiktoken einops transformers_stream_generator==0.0.4 scipy torchvision pillow tensorboard matplotlib # additional package required for Qwen-VL-Chat to conduct generation
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md:19]: Let's remove `transformers==4.31.0` here

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
+# Qwen-VL
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md:1]: For the places where we mention the general model family, let's use Qwen-VL; while when we mention the specific model in this model family, let's use Qwen-VL-Chat

Comment (Oscilloscope98) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md:1]: So here, we should have

`For illustration purposes, we utilize the [Qwen/Qwen-VL-Chat](https://huggingface.co/Qwen/Qwen-VL-Chat) as a reference Qwen-VL model.`

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+- `--repo-id-or-model-path`: str, argument defining the huggingface repo id for the Qwen-VL model to be downloaded, or the path to the huggingface checkpoint folder. It is default to be `'Qwen/Qwen-VL'`.
+- `--n-predict`: int, argument defining the max number of tokens to predict. It is default to be `32`.
+  
+In every session, image and text can be entered into cmd (user can skip the input by type **'Enter'**) ; please type **'exit'** anytime you want to quit the dialouge.
+
+Every image output will be named as the round of session and placed under the current directory.
+
+#### 2.4 Sample Chat
+#### [Qwen/Qwen-VL](https://huggingface.co/Qwen/Qwen-VL)
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md:58]: Same here, use Qwen-VL-Chat

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+ Please input a picture: 
+ Please enter the text: 在图中检测框出猫猫的墨镜
+---------- Response ----------
+<ref>猫猫的墨镜</ref><box>(398,313),(994,506)</box> 
+
+-------------------- Session 4 --------------------
+ Please input a picture: exit
+```
+
+And the input and output picture is like:
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md:83]:
```
The sample input image in Session 1 is:

<a href="https://llm-assets.readthedocs.io/en/latest/_images/qwen-vl-example-input.jpg"><img width=xxx src="https://llm-assets.readthedocs.io/en/latest/_images/qwen-vl-example-input.jpg" ></a>

The sample output image in Session 3 is:

<a href="https://llm-assets.readthedocs.io/en/latest/_images/qwen-vl-example-output.png"><img width=xxx src="https://llm-assets.readthedocs.io/en/latest/_images/qwen-vl-example-output.png" ></a>
```

Please change xxx to suitable image width.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+from transformers.generation import GenerationConfig
+import torch
+import time
+import os
+import argparse
+from bigdl.llm import optimize_model
+torch.manual_seed(1234)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Predict Tokens using `chat()` API for Qw-VL-Chat model')
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py:28]: Qw-VL-Chat -> Qwen-VL

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+    parser.add_argument('--repo-id-or-model-path', type=str, default="Qwen/Qwen_VL_Chat",
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py:29]: wrong default repo id here, which should be Qwen/Qwen-VL-Chat

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+                        help='The huggingface repo id for the Qwen_VL_Chat model to be downloaded'
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py:30]: Qwen_VL_Chat -> Qwen-VL

File: python/llm/example/CPU/PyTorch-Models/Model/qwen-vl/README.md [hunk 16]
```diff
@@ -0,0 +1,88 @@
+# Qwen-VL
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/qwen-vl/README.md:1]: Similar edit suggestion in the following 2 files

**Comment by plusbang** [comment]
Please also update links in https://github.com/intel-analytics/BigDL/tree/main/python/llm/example/CPU/HF-Transformers-AutoModels/Model and https://github.com/intel-analytics/BigDL/tree/main/python/llm/example/CPU/PyTorch-Models/Model.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+                             ', or the path to the huggingface checkpoint folder')
+    parser.add_argument('--n-predict', type=int, default=32, help='Max tokens to predict')
+    
+    current_path = os.path.dirname(os.path.abspath(__file__))
+    args = parser.parse_args()
+    model_path = args.repo_id_or_model_path  
+        
+    # Load model
+    # For successful BigDL-LLM optimization on Qwen-VL-Chat, skip the 'c_fc' and 'out_proj' modules during optimization
+    model = AutoModelForCausalLM.from_pretrained(model_path, load_in_4bit=True, device_map="cpu", trust_remote_code=True, optimize_model=True,modules_to_not_convert=['c_fc', 'out_proj'] )
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py:40]: Maybe also change the alignment here:

```python
model = AutoModelForCausalLM.from_pretrained(model_path, 
                                             load_in_4bit=True, 
                                             device_map="cpu",
                                             trust_remote_code=True, 
                                             modules_to_not_convert=['c_fc', 'out_proj'] )
``` 

Also `optimize_model` arg can be removed here as it is True by default.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+-------------------- Session 3 --------------------
+ Please input a picture: 
+ Please enter the text: 在图中检测框出猫猫的墨镜
+---------- Response ----------
+<ref>猫猫的墨镜</ref><box>(398,313),(994,506)</box> 
+
+-------------------- Session 4 --------------------
+ Please input a picture: exit
+```
+The sample input image in Session 1 is:
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/README.md:82]: The sample input image in Session 1 is (which is referred from xxxxx):

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+from transformers.generation import GenerationConfig
+import torch
+import time
+import os
+import argparse
+from bigdl.llm import optimize_model
+torch.manual_seed(1234)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Predict Tokens using `chat()` API for Qw-VL model')
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/qwen-vl/chat.py:28]: Qwen-VL

File: python/llm/example/CPU/PyTorch-Models/Model/qwen-vl/chat.py [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+from transformers.generation import GenerationConfig
+import torch
+import time
+import os
+import argparse
+from bigdl.llm import optimize_model
+torch.manual_seed(1234)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Predict Tokens using `chat()` API for Qw-VL model')
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/qwen-vl/chat.py:27]: Qwen-VL

File: python/llm/example/CPU/PyTorch-Models/Model/qwen-vl/README.md [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+ Please input a picture: 
+ Please enter the text: 在图中检测框出猫猫的墨镜
+---------- Response ----------
+<ref>猫猫的墨镜</ref><box>(398,313),(994,506)</box> 
+
+-------------------- Session 4 --------------------
+ Please input a picture: exit
+```
+
+The sample input image in Session 1 is:
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/qwen-vl/README.md:83]: Same here

**Review by Oscilloscope98 — APPROVED** [review]
LGTM

**Pull request merged**